src/Apps/CollectorProfile/Routes/Saves2/Components/Actions/EditSavesModal.tsx

```diff
@@ -0,0 +1,153 @@
+      .max(MAX_NAME_LENGTH),
+  })
+
+  const { submitMutation } = useUpdateCollection()
+
+  const { sendToast } = useToasts()
+
+  const handleSubmit = async (
+    formikValues: FormikValues,
+    formikHelpers: FormikHelpers<FormikValues>
+  ) => {
+    try {
+      await submitMutation({
+        variables: {
+          input: {
+            id: collection.internalID,
+            name: formikValues.name,
+          },
+        },
+        rejectIf: res => {
+          const result = res.updateCollection?.responseOrError
+
+          return result?.__typename === "UpdateCollectionFailure"
+            ? result.mutationError
+            : false
+        },
+      })
+
+      sendToast({
+        variant: "success",
+        message: t("collectorSaves.editListModal.success"),
+      })
+    } catch (error) {
+      logger.error(error)
+      sendToast({
+        variant: "error",
+        message: error.message ?? t("common.errors.somethingWentWrong"),
+      })
+    } finally {
+      formikHelpers.setSubmitting(false)
+      setIsEditModalOpen(false)
+    }
+  }
+
+  return (
+    <Formik<FormikValues>
+      initialValues={initialValues}
+      validationSchema={validationSchema}
+      onSubmit={handleSubmit}
+    >
+      {({
+        values,
+        errors,
+        touched,
+        handleBlur,
+        handleChange,
+        isSubmitting,
+      }) => {
+        return (
+          <ModalDialog
+            title={t("collectorSaves.editListModal.title")}
+            onClose={() => {
+              setIsEditModalOpen(false)
+            }}
+            width={["100%", 700]}
+          >
+            <Form /* TODO: confirm implicit form submission */>
+              <LabeledInput
+                name="name"
+                value={values.name}
+                title={t("collectorSaves.editListModal.fields.name.label")}
+                label={<EditIcon />}
+                error={touched.name && errors.name}
+                maxLength={MAX_NAME_LENGTH}
+                required // TODO: confirm
+                onChange={handleChange}
+                onBlur={handleBlur}
+              />
+              <Spacer y={1} />
+              <Text variant="xs">
+                {t(
+                  "collectorSaves.createNewListModal.remainingCharactersCount",
+                  { count: MAX_NAME_LENGTH - values.name.length }
+                )}
+              </Text>
+              <Spacer y={2} />
+              <Flex
+                justifyContent={["space-between"]}
+                flexDirection={["column", "row-reverse"]}
+              >
+                <Button type="submit" mb={[1, 0]} loading={!!isSubmitting}>
+                  {t("collectorSaves.editListModal.save")}
+                </Button>
+
+                <Button
+                  variant="secondaryBlack"
+                  onClick={() => setIsEditModalOpen(false)}
+                  display={["block", "none"]} // TODO: intentionally hidden on desktop?
```

Comment:
As far as I can see by design, then yes, this is the desired behavior when this button will be hidden for desktop
<img width="883" alt="Untitled – Figma 2023-03-03 14-09-00" src="https://user-images.githubusercontent.com/3513494/222705470-7e05b6d1-4f8f-4873-a149-e7ee67edd742.png">

Reply:
I'm confirming [here](https://artsy.slack.com/archives/C9SATFLUU/p1677803444831519?thread_ts=1677803388.476209&cid=C9SATFLUU) (since this seems to differ from the Delete flow)